File: assets/Scripts/vidas.cs
using UnityEngine;
using System.Collections;

public class vidas : MonoBehaviour {
	public int vida;
	public GameObject vida1 ;
	public GameObject vida2 ;
	public GameObject vida3 ;
	public GUIStyle CONTINUARStyle;
	public GUIStyle REINICIARStyle;
	public GUIStyle SALIRStyle;
	public bool Mostrar = false;

	public GameObject Herramienta ;
	private Component Mover;
	private ClickAndGo click;


	// Use this for initialization
	void Start () {
		Mover = Herramienta.GetComponent<ClickAndGo>();

	}
	
	// Update is called once per frame
	void Update () {

		if (vida >3){ 
			vida = 3; 
		} 
		 
		if (vida == 2){ 
			vida3.SetActive(false);
		} 
		if (vida == 1){ 
			vida2.SetActive(false);
		} 
		if (vida == 0){ 
			
			//(Mover as MonoBehaviour).enabled = false;
			Time.timeScale =  0f;
			vida1.SetActive(false);
			Herramienta.SetActive(false);
			Mostrar = true;
			OnGUI();
		} 
	}
	
	void OnTriggerEnter2D(Collider2D col){
		if(col.tag == "Enemy")
		{
			vida-=1;
			Update();
		}
		//if (col.tag == "Obstacle") {
		//	Time.timeScale = 0;
		//	Herramienta.SetActive(false);
		//	pausa=true;
		//	Mostrar = true;
		//}
	}
 

	void OnGUI(){
		GUI.matrix = Matrix4x4.TRS(Vector3.zero, Quaternion.identity,new Vector3(Screen.width / 480.0f, Screen.height / 320.0f, 1));
		if(Mostrar == true)
		{


			if(GUI.Button(new Rect(80,100,150,100),"",REINICIARStyle))
			{
					Mostrar = false;
					Time.timeScale = 1f;
					Application.LoadLevel("level1");
					vida = 3;
			}

			if(GUI.Button(new Rect(260,100,150,100),"",SALIRStyle))
			{
				Application.LoadLevel("Menu");
				
			}
		}
	}
}

File: assets/Scripts/Level One/vidas.js
#pragma strict

import UnityEngine.SceneManagement;

	public var  vida:int;
	public var  desing:GUISkin;

	public var  Mostrar:boolean = false;

	public var  Herramienta:GameObject ;
	private var  Mover:Component;
	private var click:clickAndGo;
	public var posicionInicial:Vector3;
	
	private var ui:GameObject[];
	public var level:int;
	private var fondo:GameObject;
	
	public static var life3:Sprite;
	public var life2:Sprite;
	public var life1:Sprite;
	public var life0:Sprite;
	static var actualLife = 3;
	public var barraVidas:GameObject;
	var windowRect:Rect;

	public var mostrado = false;


	// Use this for initialization
	function Start () {

		ui = new GameObject[9];
		Mover = Herramienta.GetComponent(clickAndGo);


		ui = GameObject.FindGameObjectsWithTag("ui");
		fondo = GameObject.Find("Fondo");
		if(level!=2)
			posicionInicial = Herramienta.transform.position;

		cargar();

	}
	
	// Update is called once per frame
	function Update () {

		if (vida >3){ 
			vida = 3; 
			barraVidas.GetComponent(UnityEngine.UI.Image).sprite = life3;
		} 
		 
		if (vida == 2){ 
			barraVidas.GetComponent(UnityEngine.UI.Image).sprite = life2;
			actualLife = 2;
		} 
		if (vida == 1){ 
			barraVidas.GetComponent(UnityEngine.UI.Image).sprite = life1;
			actualLife = 1;
		} 
		if (vida == 0 && mostrado == false){ 
			
			barraVidas.GetComponent(UnityEngine.UI.Image).sprite = life0;
			//(Mover as MonoBehaviour).enabled = false;
			Time.timeScale =  0f;
			
			//Herramienta.SetActive(false);
			Mostrar = true;

			settingUi(false);
			mostrado = true;
			//OnGUI();
		} 
	}
	
	function OnTriggerEnter2D(col:Collider2D){
		if(col.tag == "Enemy")
		{
			vida-=1;
			//Update();
		}
		
	}
	
 	function func(){
 		
				
		GUILayout.BeginHorizontal ();
		if (GUILayout.Button ("Reiniciar")) {

			
			Mostrar = false;
			SceneManager.LoadScene("level"+level);
			settingUi(true);


			mostrado = false;
			
		}
		if (GUILayout.Button ("Salir")) {
			Mostrar = false;
			if(level == 0)
				SceneManager.LoadScene("menuInicial");
				//Application.LoadLevel("menuInicial");
			else
				SceneManager.LoadScene("Menu");
				//Application.LoadLevel("Menu");	
			settingUi(true);
			
		}
		
		GUILayout.EndHorizontal ();
		
						
	}

	function OnAwake(){
		
	}
	
	function OnGUI(){
		GUI.skin = desing;	
		if (Mostrar) {
			windowRect = GUI.Window(0,windowRect,func,"Has Perdido \n");
			windowRect = new Rect (Screen.width / 2 -220, Screen.height / 2 -100, 500, 100);
		}
	
	}

	function settingUi(bol)
	{
		var bole:boolean;
		Debug.Log("Bole"+ bole);
		bole = bol;

		Debug.Log(level);
		//gameObject.renderer.enabled = bole;
		
		if(level == 0)
		{
			gameObject.GetComponent.<GUITexture>().enabled = bole;
		}
		
		if(level == 2 || level == 1 )
		{
			//	gameObject.renderer.enabled = bole;
			Herramienta.GetComponent(movimientoContinuo).posicion = posicionInicial;
			
		}	
		Herramienta.SetActive(bole);
		if(bole == false)
		{
		Time.timeScale = 0f;
		fondo.GetComponent(SpriteRenderer).color = Color.gray;
		}
		else
		{
			fondo.GetComponent(SpriteRenderer).color = Color.white;
			Time.timeScale = 1f;
		}

		for(var i=0;i<ui.Length;i++)
		{
			//Debug.Log(ui[i].name);
			//print (ui.Length);
			
			if(ui[i].GetComponent.<Renderer>() != null)
				ui[i].GetComponent.<Renderer>().enabled = bole;
			
				
			if(ui[i].name.Equals("Points")||ui[i].name.Equals("Canvas_menu"))	
				ui[i].SetActive(bole);


			if(ui[i].name.Equals("Zancudo"))	
				ui[i].SetActive(bole);
		}


	
		
	}	

	function cargar()
	{
		if(actualLife == 1){
			barraVidas.GetComponent(UnityEngine.UI.Image).sprite = life1;
		}
		else if (actualLife == 2){
			barraVidas.GetComponent(UnityEngine.UI.Image).sprite = life2;
		}


	}